src/SurveyService.Validation/Question/CreateQuestionRequestValidator.cs

```diff
@@ -0,0 +1,50 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.SurveyService.Data.Interfaces;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Question;
+using LT.DigitalOffice.SurveyService.Models.Dto.Requests.Question.Filters;
+using LT.DigitalOffice.SurveyService.Validation.Option.Interfaces;
+using LT.DigitalOffice.SurveyService.Validation.Question.Interfaces;
+using System;
+
+namespace LT.DigitalOffice.SurveyService.Validation.Question;
+
+public class CreateQuestionRequestValidator : AbstractValidator<CreateSingleQuestionRequest>, ICreateQuestionRequestValidator
+{
+  public CreateQuestionRequestValidator(
+    IGroupRepository _groupRepository
+    )
+  {
+    RuleFor(q => q.Content)
+      .MaximumLength(300)
+      .WithMessage("Content is too long.");
+
+    When(q => q.Deadline.HasValue, () =>
+    {
+      RuleFor(q => q.Deadline)
+        .Must(d => d > DateTime.UtcNow.AddDays(1).AddSeconds(2))
+        .WithMessage("The deadline must be at least 24 hours from now.");
+    });
+
+    When(q => q.GroupId.HasValue, () =>
+    {
+      RuleFor(q => q)
+        .MustAsync(async (q, _) => (await _groupRepository.GetPropertiesAsync((Guid)q.GroupId)).Deadline == q.Deadline)
+        .WithMessage("The deadlines of questions in the group are not equal.");
+
+      RuleFor(q => q)
+        .MustAsync(async (q, _) => (await _groupRepository.GetPropertiesAsync((Guid)q.GroupId)).HasRealTimeResult == q.HasRealTimeResult)
+        .WithMessage("The conditions for displaying the results of the questions are different.");
+    });
+
+    When(q =>  !q.HasCustomOptions, () =>
+    {
+      RuleFor(q => q.Options)
```

Comment:
Add a check that a option is non-custom

Reply:
The options when creating a question are always non-custom. Custom options are created using the "create option endpoint"

Reply:
You're right, but the request to create a question with custom options may come from the front and it will be executed

Reply:
Done